src/authentication/components/ExecutionConfigModal.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useForm } from "react-hook-form";
+import {
+  ActionGroup,
+  AlertVariant,
+  Button,
+  ButtonVariant,
+  Form,
+  FormGroup,
+  Modal,
+  ModalVariant,
+  TextInput,
+  ValidatedOptions,
+} from "@patternfly/react-core";
+import { CogIcon, TrashIcon } from "@patternfly/react-icons";
+
+import type AuthenticatorConfigRepresentation from "keycloak-admin/lib/defs/authenticatorConfigRepresentation";
+import type AuthenticatorConfigInfoRepresentation from "keycloak-admin/lib/defs/authenticatorConfigInfoRepresentation";
+import type { ConfigPropertyRepresentation } from "keycloak-admin/lib/defs/authenticatorConfigInfoRepresentation";
+import type { ExpandableExecution } from "../execution-model";
+import { useAdminClient, useFetch } from "../../context/auth/AdminClient";
+import { useAlerts } from "../../components/alert/Alerts";
+import { HelpItem } from "../../components/help-enabler/HelpItem";
+
+export const ExecutionConfigModal = ({
+  execution,
+}: {
+  execution: ExpandableExecution;
+}) => {
+  const { t } = useTranslation("authentication");
+  const adminClient = useAdminClient();
+  const { addAlert, addError } = useAlerts();
+
+  const [show, setShow] = useState(false);
+  const [config, setConfig] = useState<AuthenticatorConfigRepresentation>();
+  const [configDescription, setConfigDescription] =
+    useState<AuthenticatorConfigInfoRepresentation>();
+
+  const { register, errors, setValue, handleSubmit } = useForm();
+
+  const setupForm = (
+    configDescription: any,
+    config?: AuthenticatorConfigRepresentation
+  ) => {
+    configDescription.properties.map(
+      (property: ConfigPropertyRepresentation) => {
+        setValue(
+          property.name!,
+          config?.config?.[property.name!] || property.defaultValue || ""
+        );
+      }
+    );
+    if (config) {
+      setValue("alias", config.alias);
+      setValue("id", config.id);
+    }
+  };
+
+  useFetch(
+    async () => {
+      const configDescription =
+        await adminClient.authenticationManagement.getConfigDescription({
+          providerId: execution.providerId!,
+        });
+      let config: AuthenticatorConfigRepresentation | undefined;
+      if (execution.authenticationConfig) {
+        config = await adminClient.authenticationManagement.getConfig({
+          id: execution.authenticationConfig,
+        });
+      }
```

Comment:
You can probably use a ternary here.
```suggestion
      const config = execution.authenticationConfig
        ? await adminClient.authenticationManagement.getConfig({
            id: execution.authenticationConfig,
          })
        : undefined;
```

Reply:
I don't think this is better

Reply:
I guess it's a matter of opinion, I prefer the ternary because it allows us to use a constant and because the type can be inferred automatically.

Reply:
Yes that is true, but I feel it's a lot more difficult to read

Reply:
I think that's valid, let's consider this resolved :ok_hand: